homeassistant/components/webmin/coordinator.py

```diff
@@ -0,0 +1,70 @@
+"""Data update coordinator for the Webmin integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from webmin_xmlrpc.client import WebminInstance
+from yarl import URL
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_CONNECTIONS,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_component import DEFAULT_SCAN_INTERVAL
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, LOGGER
+
+
+class WebminUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """The Webmin data update coordinator."""
+
+    config_entry: ConfigEntry
+    instance: WebminInstance
+    mac_address: str
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the Webmin data update coordinator."""
+        self.config_entry = config_entry
+        base_url = URL.build(
+            scheme="https" if config_entry.options[CONF_SSL] else "http",
+            user=config_entry.options[CONF_USERNAME],
+            password=config_entry.options[CONF_PASSWORD],
+            host=config_entry.options[CONF_HOST],
+            port=int(config_entry.options[CONF_PORT]),
+        )
+        self.instance = WebminInstance(
+            session=async_create_clientsession(
+                hass,
+                verify_ssl=config_entry.options[CONF_VERIFY_SSL],
+                base_url=base_url,
+            )
+        )
+        self.device_info = DeviceInfo(
+            configuration_url=base_url,
+            name=config_entry.options[CONF_HOST],
+        )
```

Comment:
Shouldn't we add identifiers?

Reply:
I set connections in `async_setup`, and from my understanding they are equivalent to identifiers

Reply:
I have added the identifiers, you are right, I misread the documentation.